dependencies/test.py

```diff
@@ -0,0 +1,146 @@
+                splited=line.split()
+                del splited[:1]
+                req+=splited
+        file.close()
+        return req
+
+
+
+    def addChildren(self, _node, dependencies, libs):
+        values={}
+        for _value in dependencies:
+            value=_value
+            if (value.startswith('py2')):
+                value = value[4:]
+            if value.lower() in map(str.lower, libs):
+                values[_value] = "ok"
+            else:
+                values[_value] = "nope"
+
+        _node.children = [node(dep+"- "+values[dep]) for dep in dependencies]
+        i=0;
+        for dep in dependencies:
+            if dep+".spec" in reqMap.keys():
+                self.addChildren(_node.children[i], reqMap[dep + ".spec"], libs)
+            i+=1
+
+
+
+    def retrieveLibs(self):
+        libs=[]
+        response = requests.get('https://pypi.python.org/pypi?:action=browse&c=533&show=all')
+        dom = lxml.html.fromstring(response.content)
+        for link in dom.xpath('//table[@class="list"]//a/text()'):  # select the url in href for all a tags(links)
+            libs.append(link)
+        file = open(self.path + "python3AvailableLibs.txt", 'w+')
+        file.write(' '.join(libs))
+        file.close()
+
+    def getLibsFromFile(self):
+        file = open(self.path+"python3AvailableLibs.txt")
+        for line in file:
+            splited=line.split()
+        file.close()
+        return splited
+
+
+    def saveToFile(self, fileName, text):
+        f = open(self.path + fileName, 'w+')
+        for i in text.keys():
+            f.write(i + ": " + ''.join([str(x) for x in text[i]]) + "\n")
+        f.close()
+
+
+    def compareDepenedencies(self, libs):
+        finalDerp={}
+        for _key in reqMap.keys():
+            key=_key
+            if(_key.startswith('py2')):
+                key=key[4:-5]
+            else:
+                key=key[:-5]
+
+            if key.lower()  in map(str.lower, libs):
+                finalDerp[_key] = "ok"
+            else:
+                finalDerp[_key] = "nope"
+            for _value in reqMap[_key]:
+                value=_value
+                if (_value.startswith('py2')):
+                    value=value[4:]
+                if value.lower()  in map(str.lower, libs) :
+                    finalDerp[_value] = "ok"
+                else:
+                    finalDerp[_value] = "nope"
+
+        return finalDerp
+
+
+    def addDepToSpec(self, libs):
+        finalDerp={}
+        for _key in reqMap.keys():
+            key = _key
+            if (_key.startswith('py2')):
+                key = key[4:-5]
+            else:
+                key = key[:-5]
+
+            if key.lower() in map(str.lower, libs):
+                finalDerp[_key] = "ok"
+            else:
+                finalDerp[_key] = "nope"
+        return finalDerp
+
+
+
+if __name__ == '__main__':
+
+    home = '/home/adelina/cern/cmsdist/'
```

Comment:
create main function and put all code into it. Here you'll just call main function.
You also need to provide input variables as script parameters rather then hardcoded ones. Please use argparse class to do that, see for example https://github.com/vkuznet/WMArchive/blob/master/src/python/WMArchive/Tools/dump2hdfs.py
it is very concise example which shows usage of argparse module, and main function which can use external parameters.